src/main/kotlin/org/funktionale/option/Option.kt

```diff
@@ -50,51 +51,59 @@ sealed class Option<out T> {
         }
     }
 
-    fun<R> map(f: (T) -> R): Option<R> {
-        return flatMap { Some(f(it)) }
+    inline fun<R> map(f: (T) -> R): Option<R> {
```

Comment:
I understand the inline but not the need to avoid `flatMap` 

What is the benefit of doing this?